src/main/java/rs/hybridit/serviceImpl/CustomUserDetailsService.java

```diff
@@ -0,0 +1,146 @@
+package rs.hybridit.serviceImpl;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.authentication.BadCredentialsException;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+
+import rs.hybridit.auth.JwtAuthenticationRequest;
+import rs.hybridit.dto.MessageDto;
+import rs.hybridit.dto.UserDto;
+import rs.hybridit.model.Admin;
+import rs.hybridit.model.Librarian;
+import rs.hybridit.model.Role;
+import rs.hybridit.model.User;
+import rs.hybridit.model.UserTokenState;
+import rs.hybridit.repository.UserRepository;
+import rs.hybridit.security.TokenHelper;
+
+@Service
+public class CustomUserDetailsService implements UserDetailsService {
+
+	protected final Log LOGGER = LogFactory.getLog(getClass());
```

Comment:
why protected..? you can use lombok - @Slf4j

Reply:
Used lombok.